idunn/utils/auth_session.py

```diff
@@ -0,0 +1,71 @@
+from time import time
+
+import logging
+import requests
+
+
+class AuthSession:
+    """
+    Helper class for HTTP sessions that need to keep an authentification token.
+    Default behavior corresponds to an OAuth2 API.
+
+    Note that at least `get_authorization_url` and `get_authorization_params`
+    need to be overriden.
+    """
+
+    def __init__(self, expiration_tolerance=10, refresh_timeout=1000):
```

Comment:
A bit confusing to have 2 parameters with different units (s, ms). At least a docstring would be welcome.

Reply:
Oh crap, that's indeed a really bad idea, I must have put it on hold and forgotten about it later :/